server/src/test/java/umm3601/user/UserControllerSpec.java

```diff
@@ -152,74 +160,136 @@ public void setupEach() throws IOException {
+  /**
+   * A little helper method that assumes that the given context
+   * body contains an array of Users, and extracts and returns
+   * that array.
+   *
+   * @param ctx the `Context` whose body is assumed to contain
+   *  an array of `User`s.
+   * @return the array of `User`s from the given `Context`.
+   */
+  private User[] returnedUsers(Context ctx) {
+    String result = ctx.resultString();
+    User[] users = javalinJackson.fromJsonString(result, User[].class);
+    return users;
+  }
 
+  /**
+   * A little helper method that assumes that the given context
+   * body contains a *single* User, and extracts and returns
+   * that User.
+   *
+   * @param ctx the `Context` whose body is assumed to contain
+   *  a *single* `User`.
+   * @return the `User` extracted from the given `Context`.
+   */
+  private User returnedSingleUser(Context ctx) {
     String result = ctx.resultString();
-    assertEquals(db.getCollection("users").countDocuments(), JavalinJson.fromJson(result, User[].class).length);
+    User user = javalinJackson.fromJsonString(result, User.class);
+    return user;
   }
 
   @Test
-  public void getUsersByAge() throws IOException {
+  public void canGetAllUsers() throws IOException {
+    // Create our fake Javalin context
+    String path = "api/users";
+    Context ctx = mockContext(path);
+
+    userController.getUsers(ctx);
+    User[] returnedUsers = returnedUsers(ctx);
 
+    // The response status should be 200, i.e., our request
+    // was handled successfully (was OK). This is a named constant in
+    // the class HttpCode.
+    assertEquals(HttpCode.OK.getStatus(), mockRes.getStatus());
+    assertEquals(
+      db.getCollection("users").countDocuments(),
+      returnedUsers.length
+    );
+  }
+
+  @Test
+  public void canGetUsersWithAge37() throws IOException {
     // Set the query string to test with
     mockReq.setQueryString("age=37");
-
     // Create our fake Javalin context
-    Context ctx = ContextUtil.init(mockReq, mockRes, "api/users");
+    Context ctx = mockContext("api/users");
 
     userController.getUsers(ctx);
+    User[] resultUsers = returnedUsers(ctx);
 
-    assertEquals(HttpURLConnection.HTTP_OK, mockRes.getStatus());
-
-    String result = ctx.resultString();
-    User[] resultUsers = JavalinJson.fromJson(result, User[].class);
-
+    assertEquals(HttpCode.OK.getStatus(), mockRes.getStatus());
     assertEquals(2, resultUsers.length); // There should be two users returned
     for (User user : resultUsers) {
       assertEquals(37, user.age); // Every user should be age 37
     }
   }
 
   /**
-   * Test that if the user sends a request with an illegal value in
-   * the age field (i.e., something that can't be parsed to a number)
-   * we get a reasonable error code back.
-   */
+  * Test that if the user sends a request with an illegal value in
+  * the age field (i.e., something that can't be parsed to a number)
+  * we get a reasonable error code back.
+  */
```

Comment:
Quick whitespace fix 🙂

```suggestion
   * Test that if the user sends a request with an illegal value in
   * the age field (i.e., something that can't be parsed to a number)
   * we get a reasonable error code back.
   */
```